### ISSUE TITLE
Add Gradient Decent + Cost methods

### ISSUE THREAD

**Issue by marcsantiago** [sajari/regression#16]
How is the regression being tested, hyper-parameters such as the learning rate don't seem to be things that can be passed in.  How do we know the model is correct without calculating the cost?  How does it take loss into consideration?  Can the regression line be optimized to factor in gradient decent?
Cost formula 
// cost = 1/N * sum((y - (m*x+c))^2)

gradient decent formulas
// cost = 1/N * sum((y - (m*x+c))^2)
// cost/dm = 2/N * sum(-x * (y - (m*x+c)))
// cost/dc = 2/N * sum(-(y - (m*x+c)))

Which is reference in 

https://www.youtube.com/watch?v=ZPd_fKyrX48

**Comment by mish15** [comment]
It doesn't use gradient descent. It uses QR decomposition
